main.py

```diff
@@ -90,15 +87,167 @@ def handleMainWindowEvents(event, values):
+        except:
+            return
+    elif event2.split(':')[0] == '-SINGLE-SPECULAR-GREEN-EDIT-':
+        try:
+            value = int(values2[event2])
+            gvars.editedTexture.customValues[int(event2.split(':')[1])]['specular'][1] = value
+            gvars.second_window['-SINGLE-SPECULAR-SPECULAR-:' + event2.split(':')[1]].update(background_color='#' + ('{:02X}' * 3).format(gvars.editedTexture.customValues[int(event2.split(':')[1])]['specular'][0], value, gvars.editedTexture.customValues[int(event2.split(':')[1])]['specular'][2]))
+        except:
+            return
+    elif event2.split(':')[0] == '-SINGLE-SPECULAR-BLUE-EDIT-':
+        try:
+            value = int(values2[event2])
+            gvars.editedTexture.customValues[int(event2.split(':')[1])]['specular'][2] = value
+            gvars.second_window['-SINGLE-SPECULAR-SPECULAR-:' + event2.split(':')[1]].update(background_color='#' + ('{:02X}' * 3).format(gvars.editedTexture.customValues[int(event2.split(':')[1])]['specular'][0], gvars.editedTexture.customValues[int(event2.split(':')[1])]['specular'][1], value))
+        except:
+            return
+    elif event2.split(':')[0] == '-SINGLE-COLOR-RED-EDIT-':
+        try:
+            value = int(values2[event2])
+            gvars.editedTexture.customValues[int(event2.split(':')[1])]['color'][0] = value
+            gvars.second_window['-SINGLE-SPECULAR-COLOR-:' + event2.split(':')[1]].update(background_color='#' + ('{:02X}' * 3).format(value, gvars.editedTexture.customValues[int(event2.split(':')[1])]['color'][1], gvars.editedTexture.customValues[int(event2.split(':')[1])]['color'][2]))
+        except:
+            return
+    elif event2.split(':')[0] == '-SINGLE-COLOR-GREEN-EDIT-':
+        try:
+            value = int(values2[event2])
+            gvars.second_window['-SINGLE-SPECULAR-COLOR-:' + event2.split(':')[1]].update(background_color='#' + ('{:02X}' * 3).format(gvars.editedTexture.customValues[int(event2.split(':')[1])]['color'][0], value, gvars.editedTexture.customValues[int(event2.split(':')[1])]['color'][2]))
+            gvars.editedTexture.customValues[int(event2.split(':')[1])]['color'][1] = value
+        except:
+            return
+    elif event2.split(':')[0] == '-SINGLE-COLOR-BLUE-EDIT-':
+        try:
+            value = int(values2[event2])
+            gvars.editedTexture.customValues[int(event2.split(':')[1])]['color'][2] = value
+            gvars.second_window['-SINGLE-SPECULAR-COLOR-:' + event2.split(':')[1]].update(background_color='#' + ('{:02X}' * 3).format(gvars.editedTexture.customValues[int(event2.split(':')[1])]['color'][0], gvars.editedTexture.customValues[int(event2.split(':')[1])]['color'][1], value))
+        except:
+            return

```

Comment:
We could probably turn this into a single event (regex matching on event2 or changing event names and values)